tex/polyglossia.lua

```diff
@@ -110,8 +112,19 @@ local function newloader(langentry)
             for k, v in pairs(langdata) do
 				s = s .. "\n" .. k .. "\t" .. tostring(v)
             end
-            newloader_max_langid = newloader_max_langid + 4
-            langobject = lang.new(newloader_max_langid)
+            -- get next \newlanguage allocation number
+            local langcnt = tex.count[lang_register] + 1
+            -- get new lang object
+            local langobject = lang.new()
+            local langid = lang.id(langobject)
+            -- get bigger one between \newlanguage and new lang obj id
+            local maxlangid = math.max(langcnt, langid)
```

Comment:
Why ? If latex allaocated language say 20 and we get language number 19, does taking the max is safe ?

Reply:
Of course, this situation can happen. In the case you have supposed, `20` will be selected as our new language id. This is safer than selecting `19`. 
The only thing we can hope seems to be that other package writers also would honor the `\newlanaguage` register number while using `lang.new()` function. Anyway we cannot guard against every possibility of unruly package writers.

Reply:
this should be documented as comment. And we should document your reasonable expectation.